{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Supplemental Methods**\n",
    "========================"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Patients with proven CD diagnosed for at least 3 months and in clinical\n",
    "remission were recruited at the University of North Carolina (see Table\n",
    "1). Serial stool samples as outlined below were submitted for analysis.\n",
    "Exclusion criteria for entry into the study were active CD as defined by\n",
    "a short CDAI score &gt; 150; fistulizing CD, concomitant use of\n",
    "azathioprine (AZA), 6-mercaptopurine (6-MP), methotrexate or anti-TNF\n",
    "agents for less than 3 months; or concomitant use of systemic steroids\n",
    "or budesonide. Steroids or budesonide had to be discontinued at least 8\n",
    "weeks before inclusion, and local, rectally administered therapies\n",
    "containing 5-ASA (enemas, suppositories) or steroid enemas/foams should\n",
    "have not been used for the previous 4 weeks. Also, non-steroidal\n",
    "inflammatory medications (NSAIDs) were not allowed as regular treatment,\n",
    "which was defined as use for at least 4 days a week each month. Patients\n",
    "were excluded if they were on antibiotic therapy ≥5 days each month or\n",
    "had antibiotic therapy ≥5 days in the previous 24 weeks. No probiotics\n",
    "were allowed in the last 24 weeks before inclusion and patients on\n",
    "long-term therapy with narcotics &gt;2 days weekly were excluded.\n",
    "Further subject exclusions were known Hepatitis B, Hepatitis C or PSC or\n",
    "regular, high dose alcohol consumption (more than seven drinks per\n",
    "week). All trial participants were prohibited from consuming specific\n",
    "diets (e.g. Atkins diet, low carbohydrate diet). In the second subset we\n",
    "also included nonaffected family members to collect serial fecal\n",
    "samples. The requirement for the family subjects was one sibling age ≥ 8\n",
    "years without known CD or ulcerative colitis and two living parents\n",
    "without CD or ulcerative colitis. The same exclusion criteria for\n",
    "antibiotics, probiotics, narcotics and concomitant diseases as in the\n",
    "main study were applied.\n",
    "\n",
    "|  Summary                                   |  Count |\n",
    "|------|------|\n",
    "|  Male/Female (n)                                   |  8 / 11|\n",
    "|  Age (years, median; range)                        |  31 (15-51)|\n",
    "|  Duration of Crohn’s disease (years median; range) |  9 (0.5-35)|\n",
    "|  Location of Crohn’s disease (n)                   |  |\n",
    "|  Ileal                                             |  5|\n",
    "|  Ileo-colonic                                      |  10|\n",
    "|  Colonic                                           |  4|\n",
    "|  History of ileocecal resection                    |  7|\n",
    "|  History of isolated small intestinal resection    |  2|\n",
    "|  Concomitant therapies (n)                          | |\n",
    "|  Steroids                                           | 0 |\n",
    "|  Azathioprine/6-MP/methotrexate                     | 8|\n",
    "|  anti-TNF agent                                     | 12|\n",
    "|  5-ASA                                              | 10 |\n",
    "|  CDAI end of week 2 (mean; SD)                      | 63 (51)|\n",
    "|  CDAI end of week 4 (mean; SD)                      | 72 (61)|\n",
    "\n",
    "Table 1: Summary of the demographic data of the 19 CD patients.\n",
    "\n",
    "### Fecal collection\n",
    "\n",
    "Stool was collected daily using a swab technique (1), which enables the\n",
    "study subject to collect stool samples from the toilet paper. Samples\n",
    "were stored on a -20˚C freezer. Fifteen patients collected daily stool\n",
    "for two 2-week periods separated by an interval of 4 weeks, during which\n",
    "no stool was collected. The short CDAI was evaluated at entry and at the\n",
    "end of collection period 1 and 2 (2). The collection periods for the\n",
    "family substudy, which included 4 patients with inactive CD with same\n",
    "exclusion criteria as above and 3 unaffected family members of each CD\n",
    "patient were two separate 4 weeks periods interrupted by a 3 month\n",
    "collection-free interval.\n",
    "\n",
    "### DNA Extraction\n",
    "\n",
    "Fecal DNA isolation was performed according to the 16S Earth Microbiome\n",
    "Project Protocol (3). Of the 960 samples analyzed, the initial subset\n",
    "(384 samples) were processed using the Illumina MiSeq platform (150\n",
    "nucleotide sequences), and for the second subset (576 samples) were\n",
    "processed using the HiSeq platform (100 nucleotide sequences).\n",
    "\n",
    "### Data Processing\n",
    "\n",
    "Demultiplexed sequences were filtered using QIIME’s (Quantitative\n",
    "Insights Into Microbial Ecology) (4) default parameters for quality\n",
    "control using (version 1.8.0-dev). To account for the fact that a subset\n",
    "of the sequences were processed using a MiSeq instrument and the rest\n",
    "using a HiSeq instrument, the resulting reads were trimmed to an even\n",
    "length of 99 nucleotides (5, 6). The resulting sequences were clustered\n",
    "using the closed-reference OTU-picking protocol (utilizing UCLUST as the\n",
    "clustering algorithm (7)) against the Greengenes database (release\n",
    "13\\_8) (8). Distance matrices were calculated using the UniFrac (9)\n",
    "distance metric. Bimodality tests were performed using Hartigan’s dip\n",
    "test (10). To account for uneven sampling efforts in the samples, the\n",
    "resulting OTU table was rarefied at 7,400 sequences per sample. The\n",
    "number was selected so the main category of interest (health status of\n",
    "the patient) had a balanced number of classes.\n",
    "\n",
    "### Classifier\n",
    "\n",
    "To assess the benefits of increased number of samples per subject, we\n",
    "used a Random Forests (11) model and we created the features following\n",
    "these steps. First we randomly selected *N* samples for each subject.\n",
    "For this set of *N* samples, we created features based on alpha\n",
    "diversity, beta diversity and microbial relative abundances. We then\n",
    "split the dataset into training and testing subsets and evaluated the\n",
    "classifier’s sensitivity and specificity. Finally, we repeated the\n",
    "selection of *N* random samples for *M* iterations, and summarized the\n",
    "results using a receiver operating characteristic (ROC) curve and the\n",
    "area under the curve (AUC). After the *M* iterations, we increased *N*\n",
    "by one and repeated the procedure until we reached *N’*. The result is a\n",
    "total of *N’-N* ROC curves and AUC scores (one for each number of\n",
    "samples per subject).\n",
    "\n",
    "The features are created based on longitudinal patterns of the different\n",
    "data types (alpha diversity, beta diversity and relative abundances). In\n",
    "all cases, we relied on the sample collection date to order the data and\n",
    "measured a series of summary statistics. For alpha diversity and the\n",
    "microbial relative abundances, the per-sample summaries were ordered and\n",
    "treated as vectors. For beta diversity, we considered only the distances\n",
    "between ordered and subsequent timepoints and treated this as a vector.\n",
    "Features were extracted from these vectors as described in the TSFRESH\n",
    "package (12), for example by counting the number of samples where an\n",
    "individual OTU is not zero, to include the mean of alpha or beta\n",
    "diversity distances, the mean rate of change of the alpha diversity\n",
    "vector, etc.\n",
    "\n",
    "In the case of the microbial abundances, we performed a feature\n",
    "selection step using phylofactor (13). Phylofactorization was performed\n",
    "by maximizing the F-statistic from logistic regression on disease state\n",
    "in patients with CD. Over 200 factors were found as significant\n",
    "predictors of CD in the terminal ileum, even when controlling for\n",
    "multiple comparisons by a Bonferroni correction. Two factors identified\n",
    "large clades (&gt;100 OTUs) used for feature selection. Factor 3\n",
    "identified a monophyletic clade of 518 OTUs in the Lachnospiraceae\n",
    "family which decreases in the terminal ileum of patients with CD\n",
    "relative to healthy patients, and factor 26 identified a monophyletic\n",
    "clade containing 737 OTUs of the Gammaproteobacteria and\n",
    "Betaproteobacteria classes which increases in relative abundance in the\n",
    "terminal ileum of patients with CD relative to healthy patients\n",
    "(Supplementary Figure 1). The dataset used for feature selection (14),\n",
    "was not used during the training or testing stages of the classifier\n",
    "construction.\n",
    "\n",
    "Jupyter notebooks and source code describing all the analyses in this\n",
    "paper can be found in this GitHub repository:\n",
    "https://github.com/knightlab-analyses/longitudinal-ibd.\n",
    "\n",
    "### Data Availability\n",
    "\n",
    "The sequences have been deposited on EBI and are available under the\n",
    "following accession number PRJEB23009 (ERP104742). In addition, the processed sequences\n",
    "and sample information can be found in the Qiita\n",
    "(https://qiita.ucsd.edu/study/description/2538) database under the study\n",
    "identifier 2538.\n",
    "\n",
    "#### References\n",
    "\n",
    "1\\. Costello EK, Lauber CL, Hamady M, Fierer N, Gordon JI, Knight R.\n",
    "Bacterial community variation in human body habitats across space and\n",
    "time. Science. 2009;326(5960):1694-7.\n",
    "\n",
    "2\\. Thia K, Faubion WA, Jr., Loftus EV, Jr., Persson T, Persson A,\n",
    "Sandborn WJ. Short CDAI: development and validation of a shortened and\n",
    "simplified Crohn's disease activity index. Inflamm Bowel Dis.\n",
    "2011;17(1):105-11.\n",
    "\n",
    "3\\. Consortium EMP. The Earth Microbiome Project Protocols 2015\n",
    "\\[Available from:\n",
    "<http://www.earthmicrobiome.org/emp-standard-protocols/>.\n",
    "\n",
    "4\\. Caporaso JG, Kuczynski J, Stombaugh J, Bittinger K, Bushman FD,\n",
    "Costello EK, et al. QIIME allows analysis of high-throughput community\n",
    "sequencing data. Nat Methods. 2010;7(5):335-6.\n",
    "\n",
    "5\\. Caporaso JG, Lauber CL, Walters WA, Berg-Lyons D, Huntley J, Fierer\n",
    "N, et al. Ultra-high-throughput microbial community analysis on the\n",
    "Illumina HiSeq and MiSeq platforms. ISME J. 2012;6(8):1621-4.\n",
    "\n",
    "6\\. Liu Z, DeSantis TZ, Andersen GL, Knight R. Accurate taxonomy\n",
    "assignments from 16S rRNA sequences produced by highly parallel\n",
    "pyrosequencers. Nucleic Acids Res. 2008;36(18):e120.\n",
    "\n",
    "7\\. Edgar RC. Search and clustering orders of magnitude faster than\n",
    "BLAST. Bioinformatics. 2010;26(19):2460-1.\n",
    "\n",
    "8\\. McDonald D, Price MN, Goodrich J, Nawrocki EP, DeSantis TZ, Probst A,\n",
    "et al. An improved Greengenes taxonomy with explicit ranks for\n",
    "ecological and evolutionary analyses of bacteria and archaea. ISME J.\n",
    "2012;6(3):610-8.\n",
    "\n",
    "9\\. Lozupone C, Knight R. UniFrac: a new phylogenetic method for\n",
    "comparing microbial communities. Appl Environ Microbiol.\n",
    "2005;71(12):8228-35.\n",
    "\n",
    "10\\. Hartigan JA, Hartigan PM. The Dip Test of Unimodality. 1985:70-84.\n",
    "\n",
    "11\\. Breiman L. Random Forests. Machine Learning. 2001;45(1):5-32.\n",
    "\n",
    "12\\. Christ M, Kempa-Liehr, A.W., Feindt M. Distributed and parallel time\n",
    "series feature extraction for industrial big data applications. 2017.\n",
    "\n",
    "13\\. Washburne AD, Silverman JD, Leff JW, Bennett DJ, Darcy JL, Mukherjee\n",
    "S, et al. Phylogenetic factorization of compositional data yields\n",
    "lineage-level associations in microbiome datasets. PeerJ. 2017;5:e2969.\n",
    "\n",
    "14\\. Gevers D, Kugathasan S, Denson LA, Vazquez-Baeza Y, Van Treuren W,\n",
    "Ren B, et al. The treatment-naive microbiome in new-onset Crohn's\n",
    "disease. Cell Host Microbe. 2014;15(3):382-92.\n",
    "\n",
    "## Figures\n",
    "\n",
    "<img src=\"img/supplementary1.jpg\" />\n",
    "\n",
    "**Supplementary Figure 1**. Phylofactorization of the terminal ileum\n",
    "data presented in (14) reveals two major, monophyletic clades indicative\n",
    "of CD These two factors comprise of three main taxonomic groups -\n",
    "Lachnospiraceae and Beta & Gammaproteobacteria. Their isometric\n",
    "log-ratio (ILR) abundances from phylofactorization indicate the\n",
    "Lachnospiraceae decrease in patients with CD whereas the Beta &\n",
    "Gammaproteobacteria increase in patients with CD relative to controls."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
